lib/membrane_file/common_file_behaviour.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Membrane.File.CommonFileBehaviour do
```

Comment:
I'm wondering if it would be possible to generate this behaviour out of specs in `Membrane.File.CommonFile` instead of writing it manually only for the sake of tests.

Reply:
Technically, they have already been written - `credo` wouldn't allow public functions in CommonFile to not have the specs, so in this PR I've just copied the specs from there and replaced `spec` with `callback`

The only downside is that if you modify something in `CommonFile`, you might need to update `CommonFileBehaviour`. To ensure they're in sync I considered [Hammox](https://github.com/msz/hammox) linked by `mox`'s README, but it seems to be not actively maintained

Reply:
> The only downside is that if you modify something in CommonFile, you might need to update CommonFileBehaviour

That's my point, and it's not only about modifying - if you just want to see the spec, you need to check the behaviour. And in the case of 'normal' behaviour that's usually not needed, because the callbacks are called by the party that defines the behaviour. So I think that generating the behaviour automatically would be the correct approach. This makes me recall java's MySuperClass and IMySuperClass, hatfu.

Reply:
Would putting both in one file or even nesting one module inside the other solve your problem?

Reply:
Not really, but it's a good idea 👍